### PR TITLE
JSON read only results supported.

### PR DESCRIPTION
**Pull request by brianmichalk1**
I added a very simple URL that will result in JSON formatted status.

### CODE REVIEW

**Comment by brianmichalk1** [comment]
Added so_reuseaddr socket option.

**Comment by ondrej1024** [comment]
Thanks for your contribution. That is a nice addition to the API and something I wanted to do for a long time. I will test this when I am back from vacation :tropical_drink: in about 2 weeks time.

File: src/crelay.c [hunk 1]
```diff
@@ -520,6 +521,29 @@ int process_http_request(int sock)
...
       }
+      else if (strstr(url, JSON_URL))
+      {
+         /* Send regular text wrapped in HTML headers that is suitable for OpenHab
+          * For example: JSONPATH($.Relay1)
+          * as in the following line:
+          * Switch sprinklerValve1 { http=">[ON:GET:http://192.168.10.142:8000/gpio?pin=1?status=1] >[OFF:GET:http://192.168.10.142:8000/gpio?pin=1?status=0] <[sprinklerCache:5000:JSONPATH($.Relay1)]" }
+          * */
+         send_headers(fout, 200, "OK", NULL, "text/html", -1, -1);
+         printf(fout, "{");
```
Comment (ondrej1024) [inline src/crelay.c:532]: fprintf

File: src/crelay.c [hunk 2]
```diff
@@ -520,6 +521,29 @@ int process_http_request(int sock)
...
+         for (i=FIRST_RELAY; i<=last_relay; i++)
+         {
+        	 fprintf(fout, "\"Relay%d\":", i);
+        	 if (rstate[i-1] == 0){
+        		 fprint(fout, "\"OFF\"");
```
Comment (ondrej1024) [inline src/crelay.c:537]: fprintf

File: src/crelay.c [hunk 3]
```diff
@@ -520,6 +521,29 @@ int process_http_request(int sock)
...
+        	 } else {
+        		 fprint(fout, "\"ON\"");
```
Comment (ondrej1024) [inline src/crelay.c:539]: fprintf

File: src/crelay.c [hunk 4]
```diff
@@ -520,6 +521,29 @@ int process_http_request(int sock)
...
+        	 }
+        	 if (i<last_relay) {
+        		 fprintf(fout, ",");
+        	 }
+         }
+         printf(fout, "}");
```
Comment (ondrej1024) [inline src/crelay.c:545]: fprintf

**Review by ondrej1024 — COMMENTED** [review]
Please fix the warnings as suggested in the comments.

**Comment by ondrej1024** [comment]
I'm going to merge this PR and release a new version of crelay. Thanks.

File: src/crelay.c (hunk 1 above)
Comment (brianmichalk1) [reply src/crelay.c:532]: I'm not getting email notifications on your message.  Almost end of day today, but I need to resend a new PR, I think.

**Comment by brianmichalk1** [comment]
I'll submit a new PR that addresses these issues.

**Pull request closed without merging**